packages/material-ui/src/Tabs/Tabs.js

```diff
@@ -62,71 +61,149 @@ export const styles = theme => ({
   indicator: {},
 });
 
-class Tabs extends React.Component {
-  constructor() {
-    super();
+const Tabs = React.forwardRef(function Tabs(props, ref) {
+  const {
+    action,
+    centered = false,
+    children: childrenProp,
+    classes,
+    className,
+    component: Component = 'div',
+    indicatorColor = 'secondary',
+    onChange,
+    ScrollButtonComponent = TabScrollButton,
+    scrollButtons = 'auto',
+    TabIndicatorProps = {},
+    textColor = 'inherit',
+    theme,
+    value,
+    variant = 'standard',
+    ...other
+  } = props;
+  const scrollable = variant === 'scrollable';
+  const isRtl = theme.direction === 'rtl';
+
+  warning(
+    !centered || !scrollable,
+    'Material-UI: you can not use the `centered={true}` and `variant="scrollable"` properties ' +
+      'at the same time on a `Tabs` component.',
+  );
+
+  const [mounted, setMounted] = React.useState(false);
+  const [indicatorStyle, setIndicatorStyle] = React.useState({});
+  const [displayScroll, setDisplayScroll] = React.useState({
+    left: false,
+    right: false,
+  });
+  const [scrollerStyle, setScrollerStyle] = React.useState({
+    overflow: 'hidden',
+    marginBottom: null,
+  });
+  const valueToIndex = React.useRef();
+  valueToIndex.current = new Map();
```

Comment:
Is there a different with this vs `const valueToIndex = React.useRef(new Map())` ?

Reply:
I believe that the useRef first argument is used for the initial value (https://codesandbox.io/s/laughing-chandrasekhar-grk1r).
How will we reset the map between two renders?

Reply:
If it needs to reset between renders it makes sense to do it this way. Should we not calculate it when the children prop changes instead of every render?

Reply:
`children` prop (unless explicitly memoised) is typically written inline, so it's likely to be a new reference each time anyway 🙂

Reply:
So far, we are using the `valueToIndex.current` for:
- Warning when providing a wrong value
- Finding the active tab item DOM node

Reply:
> Finding the selected tab item DOM node

Should just attach a ref to it in v5. I think we allow custom tab components which means it would be breaking. If not then we can do it now.

Reply:
Actually thinking about it, for posterity maybe it's worth adding a comment to say why you're using a `ref` and not just a `const valueToIndex = new Map()` if you want it to reset every render?

Reply:
> Should just attach a ref to it in v5.

+1 for looking into what we can improve in v5. It's something we could have implemented this way in the first place. For some reason, we didn't. To dive into why it wasn't done this way.

> maybe it's worth adding a comment to say why you're using a ref and not just a const valueToIndex = new Map() if you want it to reset every render?

Seems that we don't need the ref in the first place 🙃. I have just tried without, it's all right. Doing some more tests and committing.

Reply:
bf006b8